docs/config-overview.md

```diff
@@ -0,0 +1,31 @@
+# Terramate Configuration Overview
+
+There is a series of different configurations that can be done
+on Terramate, ranging from avoiding duplication by leveraging
+powerful code generation to flexible orchestration by allowing
+control of stacks order of execution.
+
+In order to do so, Terramate works with configuration files that
+have the suffixes:
+
+* `tm`
+* `tm.hcl`
+
+Terramate files can be found in any directory of a Terramate project and
+all the files in a single directory will be handled as the concatenation
+of all of them in a single file, forming a single **configuration**.
+
+A Terramate project is essentially any collection of Terraform code
+organized into stacks. The stacks can also reference local modules
+inside the project or remote ones.
+
+It is not a hard requirement for Terramate to work that the project uses git 
```

Comment:
Maybe say that change detection feature only supports `git` at the moment but other VCSs can be added in the future?